server/index.js

```diff
@@ -20,6 +21,13 @@ const logger = require('./shared/logger');
 const runApp = promisify(app.listen.bind(app));
 
 database.initialize()
+  .then(() => umzug.pending())
+  .then(migrations => {
+    if (migrations.length) {
+      return bluebird.reject(new Error('There are pending migrations.'));
+    }
+    return bluebird.resolve();
```

Comment:
You are inside promise chain anyway, anything you return (_or do not return_) will get wrapped inside promise 😉